web-wallet/src/routes/(app)/dashboard/transactions/__tests__/page.spec.js

```diff
@@ -1,15 +1,25 @@
-import { afterEach, describe, expect, it } from "vitest";
+import { afterEach, describe, expect, it, vi } from "vitest";
 import { cleanup, render } from "@testing-library/svelte";
 import Transactions from "../+page.svelte";
 
+global.ResizeObserver = vi.fn().mockImplementation(() => ({
+	observe: vi.fn(),
+	unobserve: vi.fn(),
+	disconnect: vi.fn()
+}));
+
+vi.useFakeTimers();
+
 describe("Dashboard", () => {
 	afterEach(cleanup);
 
 	const currentPrice = { usd: 0.5 };
 
-	it("should render the transactions page", () => {
+	it("should render the transactions page", async () => {
```

Comment:
I would have put the tests you made for `Transactions` here instead, as we are going to update that component soon to make it so that receives just an array instead of a promise.

Reply:
I'm not sure as the component renders either a Back button or View All Transactions, dependent on where the component is rendered from. So I think having all the test logic within a component-scoped spec is better in this case?

Reply:
If the code was meant to stay as it is yes, the tests for Transactions are perfect and the tests for page should be different (and out of the scope of your PR).
My remark was only to point out that you know that we are going to strip the promise logic from Transactions, hence in the future such tests will be moved here.